flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/JsonRowSerializationSchemaTest.java

```diff
@@ -179,6 +181,22 @@ public void testSerializeRowWithInvalidNumberOfFields() {
                         .failsWithException(instanceOf(RuntimeException.class)));
     }
 
+    @Test
+    public void testSerializeRowWithTypesMismatch() {
+        final TypeInformation<Row> rowSchema =
+                Types.ROW_NAMED(new String[] {"f1", "f2"}, Types.INT, Types.STRING);
+        final Row row = new Row(2);
+        row.setField(0, 1);
+        row.setField(1, 1);
+        final JsonRowSerializationSchema serializationSchema =
+                new JsonRowSerializationSchema.Builder(rowSchema).build();
+        String expectErrMessage = "Failed to serialize at field: f2";
+        assertThat(
```

Comment:
nit: It seems the `Assert.assertThat` has been deprecated. Please use `MatcherAssert.assertThat` instead.